### PR TITLE
[GEOS-10223] Support MBTiles in OGC Tiles API

### PR DESCRIPTION
**Pull request by shadab-geo**
[GEOS-10223](https://osgeo-org.atlassian.net/browse/GEOS-10223)

  

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [ ] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [x] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [x] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [x] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

File: src/community/ogcapi/ogcapi-tiles/src/main/java/org/geoserver/ogcapi/tiles/Tileset.java [hunk 1]
```diff
@@ -198,6 +203,16 @@ public Tileset(
...
     }
 
+    private void validateLimits(TileMatrixSetLimit limit, GridSubset gridSubset, int zoomLevel) {
+
+        long numTilesHeight = gridSubset.getGridSet().getGrid(zoomLevel).getNumTilesHigh();
+        long numTilesWide = gridSubset.getGridSet().getGrid(zoomLevel).getNumTilesWide();
+        if (limit.getMinTileRow() < 0) limit.setMinTileRow(0);
+        if (limit.getMinTileCol() < 0) limit.setMinTileCol(0);
+        if (limit.getMaxTileRow() > numTilesWide - 1) limit.setMaxTileRow(numTilesWide - 1);
+        if (limit.getMaxTileCol() > numTilesHeight - 1) limit.setMaxTileCol(numTilesHeight - 1);
```
Comment (dromagnoli) [inline src/community/ogcapi/ogcapi-tiles/src/main/java/org/geoserver/ogcapi/tiles/Tileset.java:213]: I think that columns rely on tilesWide and rows on tilesHeight. Could you please doublecheck?

**Review by dromagnoli — CHANGES_REQUESTED** [review]
I think that you switched rows and columns on tileWide, tileHeight usage. could you please check?

File: src/community/ogcapi/ogcapi-tiles/src/main/java/org/geoserver/ogcapi/tiles/Tileset.java (hunk 1 above)
Comment (shadab-geo) [reply src/community/ogcapi/ogcapi-tiles/src/main/java/org/geoserver/ogcapi/tiles/Tileset.java]: Ohh, Got it. I have switched them!

**Review by dromagnoli — APPROVED** [review]

**Comment by dromagnoli** [comment]
@shadab-geo could you please backport that?

**Comment by shadab-geo** [comment]
> @shadab-geo could you please backport that?

sure

**Pull request merged**